app/controllers/event_log_files_controller.rb

```diff
@@ -2,17 +2,19 @@ class EventLogFilesController < ApplicationController
   include ActionController::Live
 
   ALL_EVENTS_TYPE = "All"
-  
+
   before_filter :setup_databasedotcom_client
 
   def index
     redirect_to root_path unless logged_in?
 
     @username = session[:username]
-    if not session.has_key?("event_types")
-      session[:event_types] = get_event_types
+
+    unless metadata_cached?
```

Comment:
Single line is preferred (idiomatic) way to use unless.
e.g. `load_and_cache_elf_metadata unless metadata_cached?`